DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Net.Http;
+using System.Data.Entity;
+using Microsoft.AspNet.Identity;
+using GitHubExtension.Infrastructure.Constants;
+using GitHubExtension.Security.DAL.Identity;
+using GitHubExtension.Security.DAL.Infrastructure;
+using GitHubExtention.Preferences.WebApi.Constants;
+
+
+namespace GitHubExtention.Preferences.WebApi.Controllers
+{
+    [RoutePrefix(RouteConstants.ApiUser)]
+    public class UserController : ApiController
+    {      
+        private readonly ApplicationUserManager _userManager;
+
+        public UserController(ApplicationUserManager userManager)
+        {
+            _userManager = userManager;
+        }
+
+
+        [HttpPost]
+        [AllowAnonymous]
+        [Route(RouteConstants.Avatar)]
+        public async Task<IHttpActionResult> Post()
+        {
+            if (!Request.Content.IsMimeMultipartContent(AvatarConstants.FormData))
+            {
+                throw new HttpResponseException(HttpStatusCode.UnsupportedMediaType);
+            }
+
+            string currentUserId = User.Identity.GetUserId();
+            User user = await _userManager.Users.FirstOrDefaultAsync(x => x.Id == currentUserId);
+            if (user == null)
+                return NotFound();
+
+            var multipartStreamProvider = new AzureBlobStorageMultipartProvider(BlobHelper.GetWebApiContainer(), User.Identity.Name);
```

Comment:
Can you move this initialization to SimpleInjector? We should have only one entry point for creation objects.